streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -0,0 +1,428 @@
+                        continue;
+                    } else if (next.key.window().end() > timestamp) {
+                        if (!foundFirst) {
+                            foundFirst = true;
+                            if (isLeftWindow(next)) {
+                                foundLeftFirst = true;
+                            }
+                        }
+                        rightWinAgg = next.value;
+                        putAndForward(next.key.window(), next.value, key, value, closeTime, timestamp);
+                        continue;
+                    } else if (next.key.window().end() == timestamp) {
+                        putAndForward(next.key.window(), next.value, key, value, closeTime, timestamp);
+                        leftWinExists = true;
+                        continue;
+                    } else {
+                        if (!foundFirstEndTime) {
+                            leftWinAgg = next.value;
+                            foundFirstEndTime = true;
+                        }
+                        if (prevRightWinExists) {
+                            break;
+                        }
+                        if (isLeftWindow(next)) {
+                            prevRightWinExists = true;
+                            final long rightWinStart = next.key.window().end() + 1;
+                            if (windowStartTimes.contains(rightWinStart)) {
+                                prevRightWinAlreadyCreated = true;
+                            } else {
+                                final TimeWindow window = new TimeWindow(rightWinStart, rightWinStart + windows.timeDifference);
+                                final ValueAndTimestamp<Agg> valueAndTime = ValueAndTimestamp.make(initializer.apply(), timestamp);
+                                putAndForward(window, valueAndTime, key, value, closeTime, timestamp);
+                            }
+                        }
+                    }
+                }
+            }
+
+            //create the left window of the current record if it's not created
+            if (!leftWinExists) {
+                final Agg aggValue;
+                final long newTimestamp;
+                //confirms that the left window contains more than the current record
+                if (prevRightWinExists) {
+                    aggValue = aggregator.apply(key, value, getValueOrNull(leftWinAgg));
+                    newTimestamp = leftWinAgg.timestamp();
+                } else {
+                    //left window just contains the current record
+                    aggValue = aggregator.apply(key, value, initializer.apply());
+                    newTimestamp = timestamp;
+                }
+                final TimeWindow window = new TimeWindow(timestamp - windows.timeDifference, timestamp);
+                final ValueAndTimestamp<Agg> valueAndTime = ValueAndTimestamp.make(aggValue, Math.max(timestamp, newTimestamp));
```

Comment:
Since it's a left window, the max timestamp should always be `timestamp`, right?

Reply:
yes!